src/main/java/org/killbill/billing/plugin/adyen/client/payment/service/BaseAdyenPaymentServiceProviderPort.java

```diff
@@ -20,33 +20,130 @@
 import javax.annotation.Nullable;
 
 import org.killbill.billing.plugin.adyen.client.model.PaymentData;
+import org.killbill.billing.plugin.adyen.client.model.PaymentModificationResponse;
+import org.killbill.billing.plugin.adyen.client.model.PurchaseResult;
 import org.killbill.billing.plugin.adyen.client.model.UserData;
 import org.slf4j.Logger;
 
 public abstract class BaseAdyenPaymentServiceProviderPort {
 
-    protected void logOperation(final Logger logger, final String operation, @Nullable final PaymentData paymentData, @Nullable final UserData userData, @Nullable final String pspReference) {
-        final StringBuilder stringBuilder = new StringBuilder("op='").append(operation).append("'");
-        if (paymentData != null && paymentData.getAmount() != null) {
-            stringBuilder.append(", amount='")
-                         .append(paymentData.getAmount())
-                         .append("'");
+    protected Logger logger;
+
+    protected void logTransaction(final String transactionType, final UserData userData, final String merchantAccount, final PaymentData paymentData, @Nullable final PurchaseResult result, @Nullable final AdyenCallResult<?> adyenCall) {
+        final StringBuilder logBuffer = new StringBuilder();
+        appendTransactionType(logBuffer, transactionType);
+        appendMerchantAccount(logBuffer, merchantAccount);
+        appendPaymentData(logBuffer, paymentData);
+        appendUserData(logBuffer, userData);
+        appendPurchaseResult(logBuffer, result);
+        if (adyenCall != null) {
+            appendDuration(logBuffer, adyenCall.getDuration());
         }
-        if (paymentData != null && paymentData.getPaymentTransactionExternalKey() != null) {
-            stringBuilder.append(", paymentTransactionExternalKey='")
-                         .append(paymentData.getPaymentTransactionExternalKey())
-                         .append("'");
+        logBuffer.append(", error=false");
+
+        logger.info(logBuffer.toString());
+    }
+
+    protected void logTransaction(final String transactionType, final String pspReference, String merchantAccount, final PaymentData paymentData, final PaymentModificationResponse response, final AdyenCallResult<?> adyenCall) {
+        final StringBuilder logBuffer = new StringBuilder();
+        appendTransactionType(logBuffer, transactionType);
+        appendMerchantAccount(logBuffer, merchantAccount);
+        appendPaymentData(logBuffer, paymentData);
+        appendPspReference(logBuffer, pspReference);
+        appendModificationResponse(logBuffer, response);
+        appendDuration(logBuffer, adyenCall.getDuration());
```

Comment:
> But we could expand the result call for failures in the class if you prefer. Maybe it's more clear?

Not strong opinion but I do think indeed that it would be clearer for the reader.